examples/vector_return/src/lib.rs

```diff
@@ -0,0 +1,30 @@
+extern crate libc;
+
+use libc::size_t;
+use std::mem;
+
+#[no_mangle]
+pub extern fn counter_generate(start: size_t, size: size_t, vec: *mut *mut size_t) -> size_t {
+    let mut counted = Vec::new();
+
+    for x in start..start + size as size_t {
+        counted.push(x);
+    }
```

Comment:
Ah, I see why you used `size_t` now.

I'd prefer this to be written as

```rust
    let counted: Vec<_> = (start..).take(s).collect();
```

This has two main benefits:

1. Removes the mutability requirement.
1. Allocates the proper amount of memory for the vector all at once at the beginning.